File: fe/src/api/clientService.ts
import api from './axios';

export const clientService = {
  getAll: async (page = 0, size = 10, search = '') => {
    const res = await api.get(`/api/clients?page=${page}&size=${size}&search=${search}`);
    return res.data;
  },
  getById: async (id: string | number) => {
    const res = await api.get(`/api/clients/${id}`);
    return res.data;
  },
  create: async (data: any) => {
    const res = await api.post('/api/clients', data);
    return res.data;
  },
  update: async (id: string | number, data: any) => {
    const res = await api.put(`/api/clients/${id}`, data);
    return res.data;
  }
};

File: backend/src/main/java/com/raghav/backend/service/ClientService.java
package com.raghav.backend.service;

import com.raghav.backend.dto.client.ClientRequest;
import com.raghav.backend.dto.client.ClientResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ClientService {
    Page<ClientResponse> getAll(Pageable pageable);

    ClientResponse create(ClientRequest request);

    ClientResponse getById(Long id);

    ClientResponse update(Long id, ClientRequest request);

    void softDelete(Long id);
}
